claat/serve.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"archive/zip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// cmdServe is the "claat serve ..." subcommand.
+func cmdServe() {
+	var depsDir = "bower_components"
+	var elemDir = "elements"
+	var elemFile = elemDir + "/codelab.html"
+	const codelabElem = `
+<link rel="import" href="../bower_components/google-codelab-elements/google-codelab-elements.html">
+`
+	err := os.MkdirAll(depsDir, 0755)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	// Go get the dependencies.
+	err = fetchRepo(depsDir, "googlecodelabs/codelab-components#^1.0.0")
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	//os.Rename(depsDir+"/codelab-components", depsDir+"/google-codelab-elements")
+	//os.Rename(depsDir+"/code-prettify", depsDir+"/google-prettify")
+	err = os.MkdirAll(elemDir, 0755)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	if _, err := os.Stat(elemFile); os.IsNotExist(err) {
+		f, err := os.Create(elemFile)
+		if err != nil {
+			log.Fatal("Cannot create file", err)
+		}
+		defer f.Close()
+		fmt.Fprintf(f, codelabElem)
+	}
+	fmt.Println("Dependencies installed.")
+
+	port := "9090"
```

Comment:
Add it as cmd line arg, e.g. `claat serve -port=9090`

Reply:
Better yet, `claat serve -addr=localhost:9090`, so one can specify anything, `-addr=localhost:9090` being the default I guess, and you won't have to concatenate host + port manually.

Reply:
good idea, fixed.